### ISSUE TITLE
no matches for kind "Ingress" in version "networking.k8s.io/v1beta1"

### ISSUE THREAD

**Issue by lapwat** [elastic/helm-charts#1452]
I am trying to install `elastic/kibana` helm chart with custom values:

```sh
helm install -n instances kibana elastic/kibana -f values.yaml  --dry-run --version 7.15.0
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "Ingress" in version "networking.k8s.io/v1beta1"
```

values.yaml

```yml
ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: nginx
    kubernetes.io/tls-acme: "true"
    cert-manager.io/cluster-issuer: letsencrypt
  hosts:
  - host: my.domain.com
    paths:
    - path: /
tls:
- hosts:
  - my.domain.com
  secretName: certificate-domain
```

I do not understand how this error can be triggered because there is no reference to `networking.k8s.io/v1beta1` in the chart. This issue must have been resolved by #1409

**Comment by framsouza** [comment]
which Kubernetes version are you using? Make sure you are running at the least >v1.19 as per our [matrix of support.](https://github.com/elastic/helm-charts/blob/main/helpers/matrix.yml)

**Comment by lapwat** [comment]
```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:41:28Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:32:41Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by framsouza** [comment]
It doesn't seems to be a bug, I was able to reproduce it on version 1.22 without issue. Are you sure you are using the right kibana values for the [ingress session](ingress:)?  I saw you are using annotations to specify the nginx class but it's not the case anymore.

**Comment by eddiesimeon** [comment]
@lapwat - the main branch has the kibana chart changes required for k8s 1.22.

I have tested the main branch kibana chart with 1.22 and kibana works fine.

the next release v 7.16.0 of the Kibana chart will have the required changes for k8s 1.22.

**Comment by framsouza** [comment]
@eddiesimeon as you can see [here](https://github.com/elastic/helm-charts/blob/main/helpers/matrix.yml), we still do NOT support 1.22. We are work on that on the next week months. The elasticsearch release 7.16 there's nothing to do with the support of k8s 1.22.

@lapwat please, feel free to open up a discussion in our [elastic discuss](https://discuss.elastic.co) in case you are having issues to deploy it, as it doesn't seem to be bug I will close this issue.

**Comment by lapwat** [comment]
Actually it fails with those values:

```
ingress:
  enabled: true
```

I was not able to reproduce on a fresh kind cluster. Thanks for your help.

**Comment by lapwat** [comment]
For your information, when I print the chart on kind cluster (server 1.21.1) with version 7.15.0 which is the latest version.

```sh
helm install -n instances kibana elastic/kibana -f values.yml  --dry-run --version 7.15.0
```

I get the following output :

```txt
W1123 11:36:59.555649   87397 warnings.go:70] networking.k8s.io/v1beta1 Ingress is deprecated in v1.19+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
...
---
# Source: kibana/templates/ingress.yaml
apiVersion: **networking.k8s.io/v1beta1**
kind: Ingress
...
```

Which means version 7.15.0 does not contain the fix from #1409 yet.

Therefore this issue is normal when you have a cluster version 1.22+.
